src/cargo/util/toml/mod.rs

```diff
@@ -905,6 +937,22 @@ impl TomlDependency {
             cx.warnings.push(msg);
         }
 
+        if details.git.is_none() && details.registry.is_none() &&
+            details.path.is_some() {
+            match kind {
+                // `cargo [build|test|bench]` are not affected.
+                Some(Kind::Normal) => {
```

Comment:
Bug: currently we cannot know if the manifest is loaded for *cargo install* or *cargo package*. `kind` is always None for normal dependencies.

Reply:
In fact, I don't know why I'm checking this here. The original code doesn't have this logic too. :P